doc/developer-notes.md

```diff
@@ -640,6 +640,28 @@ Strings and formatting
 
   - *Rationale*: Bitcoin Core uses tinyformat, which is type safe. Leave them out to avoid confusion.
 
+- Use `.c_str()` sparingly. Its only valid use is to pass C++ strings to C functions that take NULL-terminated
```

Comment:
Could say null-terminated instead of NULL-terminated. NULL constant is technically unrelated

Reply:
It's the NULL or NUL character, too. C's NULL constant is indeed unrelated.